packages/type-utils/src/isTypeIntrinsic.ts

```diff
@@ -0,0 +1,12 @@
+import type * as ts from 'typescript';
+
+export interface IntrinsicType extends ts.Type {
+  intrinsicName: string;
+}
+
+export function isTypeIntrinsic(
+  type: ts.Type,
+  intrinsicName: string,
+): type is IntrinsicType {
+  return type.intrinsicName === intrinsicName;
```

Comment:
This actually is covered by tests, just in the `eslint-plugin` package. I wish codecov could merge monorepo reports like this (maybe it can?).

Reply:
#6701